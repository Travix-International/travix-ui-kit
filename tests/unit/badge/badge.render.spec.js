import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import Badge from '../../../components/badge/badge';

describe('Badge', () => {
  describe('#render()', () => {
    it('should render children', () => {
      const wrapper = shallow(
        <Badge title="Badge title">
          Badge Content
        </Badge>
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should not render wrapper if children is not provided', () => {
      const wrapper = shallow(
        <Badge title="Badge title" />
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
