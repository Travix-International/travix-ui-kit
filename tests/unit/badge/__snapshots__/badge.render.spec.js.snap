// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge #render() should not render wrapper if children is not provided 1`] = `
<div
  className="ui-badge-badge"
  mods={Array []}
>
  Badge title
</div>
`;

exports[`Badge #render() should render children 1`] = `
<div
  className="ui-badge"
>
  Badge Content
  <div
    className="ui-badge-badge"
    mods={Array []}
  >
    Badge title
  </div>
</div>
`;
